client/factory.go

```diff
@@ -91,5 +91,10 @@ func NewBucket(logger log.Logger, confContentYaml []byte, reg prometheus.Registe
 		return nil, errors.Wrap(err, fmt.Sprintf("create %s client", bucketConf.Type))
 	}
 
-	return objstore.NewTracingBucket(objstore.BucketWithMetrics(bucket.Name(), objstore.NewPrefixedBucket(bucket, bucketConf.Prefix), reg)), nil
+	return objstore.NewPrefixedBucket(bucket, bucketConf.Prefix), nil
```

Comment:
Hm, I think it's worth to mention in change log AND in comment of NewBucket, that this NewBucket no longer instruments and adds span middleware. Callers have to add them on their own now.